cpp/src/unary/math_ops.cu

```diff
@@ -639,6 +667,11 @@ std::unique_ptr<cudf::column> unary_operation(cudf::column_view const& input,
     case cudf::unary_operator::NOT:
       return cudf::type_dispatcher(
         input.type(), detail::LogicalOpDispatcher<detail::DeviceNot>{}, input, stream, mr);
+    case cudf::unary_operator::NEGATE:
+      CUDF_EXPECTS(detail::is_signed_numeric(input.type().id()),
+                   "NEGATE operator requires signed numeric types.");
```

Comment:
*question*: Should we allow negation of `duration` types?

Reply:
Yeah I didn't realize this before, but  one of the polars tests was failing due to this.